plugins/obs-webrtc/webrtc/src/ffi/output.rs

```diff
@@ -0,0 +1,204 @@
+use crate::output::{EncodedPacket, EncodedPacketType, OutputStream, OutputStreamError};
+use anyhow::Result;
+use log::{error, info};
+use std::{
+    os::raw::{c_char, c_void},
+    slice,
+    time::Duration,
+};
+use tokio::runtime::Runtime;
+
+pub struct OBSWebRTCWHIPOutput {
+    stream: OutputStream,
```

Comment:
it might make more sense to wrap the `OutputStream` itself in an `Arc` instead of wrapping basically every struct member of it in an `Arc`.

Reply:
Yeah, some of the webrtc apis take the Arc<> values (e.g. peer_connection.add_transceiver_from_track). I'll reorganize some of the code for potentially enabling some of the others to take advantage of that